Sources/MessagingPush/UserNotificationsFramework/README.md

```diff
@@ -0,0 +1,15 @@
+#  `UserNotifications` Framework
```

Comment:
Sometimes there are parts of our code that could use some documentation explaining some design patterns and decisions made. 

In recent PR comments, it was suggested to write a README inside of a directory in our SDK to explain code in only that directory. 

Seemed like a good idea in this scenario so I decided to try it. 

Feel free to offer suggestions below if the document's explanation is not clear to you. With my head being in this code for so long, it's hard to explain it 😆.